src/gs-updates-page.c

```diff
@@ -1799,9 +1799,11 @@ gs_updates_page_upgrade_cancel_cb (GsUpgradeBanner *upgrade_banner,
 static void
 scheduler_invalidated_cb (GsUpdatesPage *self)
 {
-	g_warning ("The Mogwai Scheduler has just been invalidated when "
-		   "it should still be held!");
-	g_assert_not_reached ();
+	/* The scheduler shouldn’t normally be invalidated, since we Hold() it
+	 * until we’re done with it. However, if the scheduler is stopped by
+	 * systemd (`systemctl stop mogwai-scheduled`) this signal will be
+	 * emitted. */
+	g_clear_pointer (&self->scheduler);
```

Comment:
I think we still would want to warn and abort on regular builds... can't you wrap this up in some sort of `#ifdef G_DEBUG` or something like that?

Reply:
I don’t think we do. While we’ve requested that the scheduler be held open, we can’t guarantee its behaviour. It’s not very robust if one process exiting causes another to abort.

Reply:
Okay. I have no other comments on this MR.